08-generic_cache/02_mutex/cache.go

```diff
@@ -0,0 +1,44 @@
+package cache
+
+import "sync"
+
+// Cache is key, value storage.
+type Cache[K comparable, V any] struct {
+	mutex sync.Mutex
+	data  map[K]V
+}
+
+// New create a new Cache with an initialised data.
+func New[K comparable, V any]() Cache[K, V] {
+	return Cache[K, V]{
+		data: make(map[K]V),
+	}
+}
+
+// Read returns the associated value for a key,
+// or ErrNotFound if the key is absent.
+func (c *Cache[K, V]) Read(key K) (V, bool) {
+	v, ok := c.data[key]
```

Comment:
I havne't checked the book, but we at least need to comment why there is no mutex here

Reply:
It definitely needs a mutex, unless it's demonstrating a bug.